### DISCUSSION TITLE
Rename the `internal/errors` package

### DISCUSSION THREAD

**Discussion by muhlemmer** [zitadel/zitadel#7001] [Ideas]
We have a `interna/errors` package which naming is a bit unfortunate. Go also has an `errors` package in the standard library since go v1.13. As that latter is getting more and more functionality, we also start using it in various places. This often leads to import name conflicts for which the developer has to find creative names for import aliasing, including the now long outdated `caos_errs`. We already ridiculed the current situation a number of time internally, but never planned a solution.

This discussion is meant to vote on a new name.

## Naming ideas

1. `errorz`, with the Z from Zitadel.
2. `zerrors`, again Z from Zitadel, derived from the [`xerrors`](https://pkg.go.dev/golang.org/x/xerrors) idea.
3. `errs`, just a shorter but unique enough version of `errors`
4. `errz`, short with Z from Zitadel.

## Rejected ideas

1. `xerrors`, already used in [`golang.org/x/xerrrors`](https://pkg.go.dev/golang.org/x/xerrors).
2. `errors`, already used in [std lib.](https://pkg.go.dev/errors)

Renaming of the package itself would be a "simple" find-and-replace action, so it doesn't require a user-story and estimation. Unfortunately it will touch almost every Go file in the repository and will introduce a large change with likely merge conflicts if there are multiple dev branches active. Planning wise this should be done at the end of sprint and merged before we start working on the new sprint. (I would say on Thursday)

To @zitadel/engineers

**Comment by stebenz** [comment]
With all the duplicated imports sometimes removed in unrelated PRs, I second that we should rename it to end this problem.

As `caos_errs` should be removed anyway, there is also `z_errs`/`z_errors` in some files, but I would prefer `zerrors` or `z_errors` for that matter.

**Reply by eliobischof** [reply]
+1 for `z_errors`

**Reply by muhlemmer** [reply]
Why the underscore? The [Go (Google) style guide](https://google.github.io/styleguide/go/decisions.html#underscores) advises against it.

**Reply by stebenz** [reply]
Just to be clear, `zerrors` would be preferred by me, just wanted to mention `z_errors` as it is use as alias in some files.

**Reply by muhlemmer** [reply]
I was replying to @eliobischof, sorry for the confusion.

> +1 for z_errors

**Reply by eliobischof** [reply]
zerrors is fine

**Reply by livio-a** [reply]
i'd prefer `zerrors` the most

> Unfortunately it will touch almost every Go file in the repository...

so we also get a lot of linting fixed 😄

**Reply by muhlemmer** [reply]
I'm also in favor of `zerrors`.

**Comment by muhlemmer** [comment]
As 4/5 backend devs have consensus on `zerrors`, we can consider that the final outcome of the discussion. I will try to open a PR with the renaming next week Thursday morning, so we do not interfere with the sprint.

**Reply by eliobischof** [reply]
This is the PR https://github.com/zitadel/zitadel/pull/7039
